analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -413,5 +413,123 @@ public void Invocation_IsNullOrEmpty_TryFinally()
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_OnlyInBranch(string throwHelperCall)
+        {
+            var code = @$"
+if (condition)
+{{
+    Tag(""Before"");
+    {throwHelperCall}
+    Tag(""Unreachable"");
+}}
+Tag(""End"");
+";
+            var validator = SETestContext.CreateCS(code, ", bool condition").Validator;
+            validator.ValidateTagOrder("Before", "End");
+            validator.ValidateExitReachCount(1);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(ThrowHelperCalls))]
+        public void Invocation_ThrowHelper_TryCatchFinally(string throwHelperCall)
+        {
+            var code = @$"
+try
+{{
+    {throwHelperCall}
+    Tag(""Unreachable"");
+}}
+catch
+{{
+    Tag(""Catch"");
+}}
+finally
+{{
+    Tag(""Finally"");
+}}
+Tag(""End"");
+";
+            var validator = SETestContext.CreateCS(code, ", bool condition").Validator;
+            validator.ValidateTagOrder("Catch", "Finally", "Finally", "End");
+            validator.ValidateExitReachCount(2);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DataRow("System.Diagnostics.CodeAnalysis.DoesNotReturn")]
+        [DataRow("JetBrains.Annotations.TerminatesProgram")]
+        public void Invocation_ThrowHelper_Attributes(string throwHelperAttribute)
+        {
+            var code = $@"
+using System;
+using System.Diagnostics;
+
+public class Sample
+{{
+    public void Test()
+    {{
+        Tag(""Before"");
+        ThrowHelper();
+        Tag(""Unreachable"");
+    }}
+
+    [{throwHelperAttribute}]
+    public void ThrowHelper()
+    {{
+        // No implementation. The attribute should drive the analysis.
+    }}
+
+    static void Tag(string name) {{ }}
+}}
+
+namespace JetBrains.Annotations
+{{
+    [Obsolete(""Use [ContractAnnotation('=> halt')] instead"")]
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class TerminatesProgramAttribute : Attribute {{ }}
+}}
+";
+#if NETFRAMEWORK
+            code += @"
+namespace System.Diagnostics.CodeAnalysis
+{
+    public sealed class DoesNotReturnAttribute : Attribute { }
+}
+";
+#endif
+            var validator = new SETestContext(code, AnalyzerLanguage.CSharp, Array.Empty<SymbolicCheck>()).Validator;
+            validator.ValidateTagOrder("Before");
+            validator.ValidateExitReachCount(0);
+            validator.ValidateExecutionCompleted();
+        }
+
+        private static IEnumerable<object[]> ThrowHelperCalls
+        {
+            get
+            {
+                yield return new object[] { @"System.Diagnostics.Debug.Fail(""Fail"");" };
+                yield return new object[] { @"Environment.FailFast(""Fail"");" };
+                yield return new object[] { @"Environment.Exit(-1);" };
+            }
+        }
```

Comment:
Isn't it easier to do just arrow property instead, returning the array directly in this case?